### ISSUE TITLE
Thoughts on supporting Octopus Deploy script packages

### ISSUE THREAD

**Issue by ryangribble** [RamblingCookieMonster/PSDeploy#77]
Hi there,

This looks like a pretty awesome tool :+1:

As well as PSGallery and Artifactory I could see a pretty compelling use case for being able to deploy scripts to an Octopus Deploy package.  As of version 3.3 Octopus supports running powershell scripts from [within a package](http://docs.octopusdeploy.com/display/OD/Standalone+scripts#Standalonescripts-Choosingwheretosourcethescript).

The use case here would be that you have source controller powershell scripts that you want Octopus to execute during a deployment.  Does this sound like something worth implementing?

**Comment by RamblingCookieMonster** [comment]
Very cool!  Haven't used it, but is the idea that you could deploy/update one of those package with PSDeploy, presumably through their API?

Either way, if it's something you would use, I'm sure others would - if you want to take a stab at implementing it go ahead!  Hit me up if you have any questions

Cheers!

**Comment by ryangribble** [comment]
Yeah I would propose psdeploy would support an octopus server as a deployment target. The easiest way to create the package and upload it, is to use their `octo.exe` tool - is it ok to take an external dependency on a cmdline tool like that? 

If not it could also be accomplished using  file compression cmdlets and then interacting through http with the octopus server but it'd be alot more ugly than just calling the tool

**Comment by RamblingCookieMonster** [comment]
Sure, that works - because these are just scripts that we build up arguments for, it won't hurt if one of them has a dependency - the chocolatey deploymenttype [does something similar](https://github.com/RamblingCookieMonster/PSDeploy/blob/master/PSDeploy/PSDeployScripts/Chocolatey.ps1#L41).

Cheers!
